File: scripts/python/04_errors.py
#!/usr/bin/env python3
"""Error handling - catch apfel's honest 501 for unsupported endpoints."""
from openai import APIStatusError, OpenAI

client = OpenAI(base_url="http://localhost:11434/v1", api_key="not-needed")

try:
    client.embeddings.create(
        model="apple-foundationmodel",
        input="apfel runs 100% on-device.",
    )
except APIStatusError as e:
    print(f"Got expected error: HTTP {e.status_code} - {e.message}")

File: scripts/nodejs/04_errors.mjs
// Error handling - catch apfel's honest 501 for unsupported endpoints.
import OpenAI from "openai";

const client = new OpenAI({ baseURL: "http://localhost:11434/v1", apiKey: "not-needed" });

try {
  await client.embeddings.create({
    model: "apple-foundationmodel",
    input: "apfel runs 100% on-device.",
  });
} catch (err) {
  if (err instanceof OpenAI.APIError) {
    console.log(`Got expected error: HTTP ${err.status} - ${err.message}`);
  } else {
    throw err;
  }
}
